src/main/java/com/google/sps/recommendation/ProcessNearestNeighbors.java

```diff
@@ -0,0 +1,125 @@
+package com.google.sps.recommendation;
+
+import org.deeplearning4j.models.embeddings.inmemory.InMemoryLookupTable;
+import org.deeplearning4j.models.embeddings.wordvectors.WordVectors;
+import org.deeplearning4j.models.embeddings.loader.WordVectorSerializer;
+import org.deeplearning4j.models.word2vec.wordstore.VocabCache;
+
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.linalg.indexing.NDArrayIndex;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.stream.IntStream;
+import com.google.sps.tool.ArrayUtil;
+import org.mapdb.*;
+
+/* This function of this file is to ONLY pre-process the nearest neigbors of 300,000 words.
+*  It will create/update a binary .db file. 
+*  
+*  Processing Time: 
+*  This took about 12 hours to process the nearest neighbors. 
+*  Performance analysis and stress testing were done while developing this algorithm.
+*  It turns out that computing the nearest neighbors for a batch size of 500 words was 
```

Comment:
Line 30 (running on a Pixelbook) should come before all results — otherwise, it seems like you're presenting these results as being optimal for all hardware, which is not true.

Reply:
good point, thanks